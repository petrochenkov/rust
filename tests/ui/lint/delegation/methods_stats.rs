#![deny(delegated_methods, delegation_pattern)]

struct F;

impl F {
    fn foo1(&self) {}
    fn foo2(&self) {}
    fn foo3(&self) {}
    fn foo4(&self) {}
}

struct S1 {
    field: F,
}

impl S1 {
    fn foo1(&self) { self.field.foo1();} //~ ERROR
}

struct S2 {
    field: F,
}

impl S2 {
    fn foo1(&self) {self.field.foo1();} //~ ERROR
    fn foo2(&self) {self.field.foo2();} //~ ERROR
}

struct S3 {
    field: F,
}

impl S3 {
    fn foo1(&self) {self.field.foo1();} //~ ERROR
    fn foo2(&self) {self.field.foo2();} //~ ERROR
    fn foo3(&self) {self.field.foo3();} //~ ERROR
}

struct S4 {
    field: F,
}

impl S4 {
    fn foo1(&self) {self.field.foo1();} //~ ERROR
    fn foo2(&self) {self.field.foo2();} //~ ERROR
    fn foo3(&self) {self.field.foo3();} //~ ERROR
    fn foo4(&self) {self.field.foo4();} //~ ERROR
}

fn main() {}
