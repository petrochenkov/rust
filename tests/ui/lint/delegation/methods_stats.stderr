error: dstats. stmts: OneWithoutTail, args_match: Same, ret_match: Same, self_arg: Value, same_name: true, has_expr_after: false.
  --> $DIR/methods_stats.rs:17:22
   |
LL |     fn foo1(&self) { self.field.foo1();}
   |     -----------------^^^^^^^^^^^^^^^^^--
   |     |                |
   |     |                callee
   |     caller
   |
note: the lint level is defined here
  --> $DIR/methods_stats.rs:1:39
   |
LL | #![deny(delegations_per_parent_stats, delegation_pattern)]
   |                                       ^^^^^^^^^^^^^^^^^^

error: dstats. stmts: OneWithoutTail, args_match: Same, ret_match: Same, self_arg: Value, same_name: true, has_expr_after: false.
  --> $DIR/methods_stats.rs:25:21
   |
LL |     fn foo1(&self) {self.field.foo1();}
   |     ----------------^^^^^^^^^^^^^^^^^--
   |     |               |
   |     |               callee
   |     caller

error: dstats. stmts: OneWithoutTail, args_match: Same, ret_match: Same, self_arg: Value, same_name: true, has_expr_after: false.
  --> $DIR/methods_stats.rs:26:21
   |
LL |     fn foo2(&self) {self.field.foo2();}
   |     ----------------^^^^^^^^^^^^^^^^^--
   |     |               |
   |     |               callee
   |     caller

error: dstats. stmts: OneWithoutTail, args_match: Same, ret_match: Same, self_arg: Value, same_name: true, has_expr_after: false.
  --> $DIR/methods_stats.rs:34:21
   |
LL |     fn foo1(&self) {self.field.foo1();}
   |     ----------------^^^^^^^^^^^^^^^^^--
   |     |               |
   |     |               callee
   |     caller

error: dstats. stmts: OneWithoutTail, args_match: Same, ret_match: Same, self_arg: Value, same_name: true, has_expr_after: false.
  --> $DIR/methods_stats.rs:35:21
   |
LL |     fn foo2(&self) {self.field.foo2();}
   |     ----------------^^^^^^^^^^^^^^^^^--
   |     |               |
   |     |               callee
   |     caller

error: dstats. stmts: OneWithoutTail, args_match: Same, ret_match: Same, self_arg: Value, same_name: true, has_expr_after: false.
  --> $DIR/methods_stats.rs:36:21
   |
LL |     fn foo3(&self) {self.field.foo3();}
   |     ----------------^^^^^^^^^^^^^^^^^--
   |     |               |
   |     |               callee
   |     caller

error: dstats. stmts: OneWithoutTail, args_match: Same, ret_match: Same, self_arg: Value, same_name: true, has_expr_after: false.
  --> $DIR/methods_stats.rs:44:21
   |
LL |     fn foo1(&self) {self.field.foo1();}
   |     ----------------^^^^^^^^^^^^^^^^^--
   |     |               |
   |     |               callee
   |     caller

error: dstats. stmts: OneWithoutTail, args_match: Same, ret_match: Same, self_arg: Value, same_name: true, has_expr_after: false.
  --> $DIR/methods_stats.rs:45:21
   |
LL |     fn foo2(&self) {self.field.foo2();}
   |     ----------------^^^^^^^^^^^^^^^^^--
   |     |               |
   |     |               callee
   |     caller

error: dstats. stmts: OneWithoutTail, args_match: Same, ret_match: Same, self_arg: Value, same_name: true, has_expr_after: false.
  --> $DIR/methods_stats.rs:46:21
   |
LL |     fn foo3(&self) {self.field.foo3();}
   |     ----------------^^^^^^^^^^^^^^^^^--
   |     |               |
   |     |               callee
   |     caller

error: dstats. stmts: OneWithoutTail, args_match: Same, ret_match: Same, self_arg: Value, same_name: true, has_expr_after: false.
  --> $DIR/methods_stats.rs:47:21
   |
LL |     fn foo4(&self) {self.field.foo4();}
   |     ----------------^^^^^^^^^^^^^^^^^--
   |     |               |
   |     |               callee
   |     caller

error: delegation methods stats. methods_count: 1, 1.
   |
note: the lint level is defined here
  --> $DIR/methods_stats.rs:1:9
   |
LL | #![deny(delegations_per_parent_stats, delegation_pattern)]
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: delegation methods stats. methods_count: 2, 1.

error: delegation methods stats. methods_count: 3, 1.

error: delegation methods stats. methods_count: 4, 1.

error: aborting due to 14 previous errors

